pass/ProteusPass.cpp

```diff
@@ -868,6 +742,135 @@ class ProteusJitPassImpl {
       replaceWithJitLaunchKernel(M, CB);
   }
 
+  FunctionCallee getJitRegisterFatBinaryFn(Module &M) {
+    FunctionType *JitRegisterFatbinaryFnTy =
+        FunctionType::get(VoidTy, {PtrTy, PtrTy, PtrTy},
+                          /* isVarArg=*/false);
+    FunctionCallee JitRegisterFatbinaryFn = M.getOrInsertFunction(
+        "__jit_register_fatbinary", JitRegisterFatbinaryFnTy);
+
+    return JitRegisterFatbinaryFn;
+  }
+
+  void instrumentRegisterFatBinary(Module &M) {
+    Function *F = nullptr;
+
+    if (!RegisterFatBinaryName)
+      return;
```

Comment:
This is a convoluted logic. Isn't there a better way to skip this code? Or, better never call this function if there is nothing todo.

Reply:
that's our alternative for for #ifdefs. Defer, refactoring